Directory.Build.props

```diff
@@ -102,6 +102,12 @@
 
     <!-- Fixes a common error in targets implementing a NoBuild mode. -->
     <BuildProjectReferences Condition=" '$(NoBuild)' == 'true' ">false</BuildProjectReferences>
+
+    <IsReferenceAssemblyProject>false</IsReferenceAssemblyProject>
+    <IsReferenceAssemblyProject Condition="$(MSBuildProjectDirectory.EndsWith('ref'))">true</IsReferenceAssemblyProject>
+    <HasReferenceAssembly>false</HasReferenceAssembly>
```

Comment:
I think this might be better to put in Directory.Build.targets (after project file evaluation). Also, I suspect `$(IsProjectReferenceProvider)` might be a good default value. It already accounts for test assets, samples, and has already been set to false in places where we don't want `<Reference>` to be used.